File: agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/TestingTelemetryEvent.kt
@file:Suppress("FunctionName", "ClassName", "unused", "EnumEntryName", "UnusedImport")
package com.sourcegraph.cody.agent.protocol_generated;

data class TestingTelemetryEvent(
  val feature: String,
  val action: String,
  val source: SourceParams,
  val timestamp: String,
  val testOnlyAnonymousUserID: String,
)

File: agent/bindings/csharp/TestingTelemetryEvent.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class TestingTelemetryEvent
  {
    [JsonProperty(PropertyName = "feature")]
    public string Feature { get; set; }
    [JsonProperty(PropertyName = "action")]
    public string Action { get; set; }
    [JsonProperty(PropertyName = "source")]
    public SourceParams Source { get; set; }
    [JsonProperty(PropertyName = "timestamp")]
    public string Timestamp { get; set; }
    [JsonProperty(PropertyName = "testOnlyAnonymousUserID")]
    public string TestOnlyAnonymousUserID { get; set; }
  }
}
